app/core/services/Placeholders.scala

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package core.services
+
+import javax.inject.Singleton
+import core.models.ocelot._
+import java.time.LocalDate
+import scala.util.matching.Regex
+import Regex._
+
+trait PlaceholderProvider {
+  val placeholders: Placeholders
+}
+
+@Singleton
+class Placeholders {
+  val TaxYearStartDay: Int = 6
+  val TaxYearStartMonth: Int = 4
+  val timescalePattern = "\\[timescale:(today|CY([\\-+]\\d+)?)(:(long|short))?\\]"
+  val timescaleRegex: Regex = s"${timescalePattern}".r
+
+  def long(date: LocalDate): Int = date.getYear
+  def short(date: LocalDate): Int = long(date) % 100
+  def CY(offset: Int, when: LocalDate): LocalDate = {
```

Comment:
I have my doubts about this method.

Reply:
fixed bug when "when" first day of tax year